### ISSUE TITLE
Fail to run "python setup.py install"

### ISSUE THREAD

**Issue by bqbn** [mozilla-services/absearch#35]
One of the steps we do to build absearch package is to run `python setup.py install` command [[1]](https://github.com/mozilla-services/svcops/blob/master/cloudformations/absearch/build_rpm#L105), and it is now failing with the following error:

```
Installed /tmp/absearch/lib/python2.7/site-packages/argparse-1.4.0-py2.7.egg
Searching for configparser
Reading https://pypi.python.org/simple/configparser/
Best match: configparser 3.7.3
Downloading https://files.pythonhosted.org/packages/4a/4d/5d4c07cd28476ecad84ea5ad43961e50b6fd74cd24b9b81113650b4de6ee/configparser-3.7.3.tar.gz#sha256=27594cf4fc279f321974061ac69164aaebd2749af962ac8686b20503ac0bcf2d
Processing configparser-3.7.3.tar.gz
Writing /tmp/easy_install-ILnxBE/configparser-3.7.3/setup.cfg
Running configparser-3.7.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-ILnxBE/configparser-3.7.3/egg-dist-tmp-PQgfLe
zip_safe flag not set; analyzing archive contents...
backports.__init__: module references __path__
Adding UNKNOWN 0.0.0 to easy-install.pth file

Installed /tmp/absearch/lib/python2.7/site-packages/UNKNOWN-0.0.0-py2.7.egg
error: Could not find required distribution configparser
```

AFAICT, our last push, thus successful build, happened on 6/27/2018, at that time, the configparser version it used was 3.5.0.

Can you please see what we can do to fix it?

**Comment by glasserc** [comment]
I took a look at this today and wasn't able to reproduce it. Running `setup.py` from inside a virtualenv (as described in the RPM spec file) succeeds without error on my machine. More to the point, `argparse` and `configparser` have both been available (built-in) since Python 2.7, so I'm not sure why they're trying to install themselves here.

The dependencies are very out of date on this package, not least of all the fact that it seems to be written for Python 2.7. I don't really understand this application well enough to port it to Python 3, but we should probably consider doing so at some point.

**Comment by bqbn** [comment]
Hmm, our build machine still has this issue, and I can re-produce it locally as well.

My local error looks like this:

```
Installed /Users/bqbn/bench/github/absearch/lib/python2.7/site-packages/argparse-1.4.0-py2.7.egg
Searching for configparser
Reading https://pypi.python.org/simple/configparser/
Best match: configparser 3.7.4
Downloading https://files.pythonhosted.org/packages/e2/1c/83fd53748d8245cb9a3399f705c251d3fc0ce7df04450aac1cfc49dd6a0f/configparser-3.7.4.tar.gz#sha256=da60d0014fd8c55eb48c1c5354352e363e2d30bbf7057e5e171a468390184c75
Processing configparser-3.7.4.tar.gz
Writing /var/folders/gj/v0kzh3gd6r78t_s11837mtvw0000gn/T/easy_install-NAdhLp/configparser-3.7.4/setup.cfg
Running configparser-3.7.4/setup.py -q bdist_egg --dist-dir /var/folders/gj/v0kzh3gd6r78t_s11837mtvw0000gn/T/easy_install-NAdhLp/configparser-3.7.4/egg-dist-tmp-6UcTvw
zip_safe flag not set; analyzing archive contents...
backports.__init__: module references __path__
Adding UNKNOWN 0.0.0 to easy-install.pth file

Installed /Users/bqbn/bench/github/absearch/lib/python2.7/site-packages/UNKNOWN-0.0.0-py2.7.egg
error: Could not find required distribution configparser
```

Here is how I reproduced it,

1. Make sure I'm using the stock Python 2.7.16 that comes with my mac.
2. Install virtualenv with `$ pip install --user virtualenv==1.10.1`
3. Checked out absearch 0.5.0 tag, and cd into the absearch directory
4. Run `~/Library/Python/2.7/bin/virtualenv .`
5. Run `./bin/pip install -U pip`
6. Run `./bin/python setup.py install`
7. Error!

@glasserc, could you let me know if the above steps can reproduce the error for you?

Also, I was able to get the `./bin/python setup.py install` to work by adding the following two lines to the setup.py file,

```
'configparser==3.5.0',
'urllib3==1.23',
```

Basically, these are the last known working versions for the above 2 packages.

I think the reason is probably because for unpinned packages, `setup.py install` tries to download their latest versions, and unfortunately many of which don't work with Python 2.7 any more.

**Comment by glasserc** [comment]
> virtualenv==1.10.1

Why this version of virtualenv? The current one is `16.6.0`; I'm using `16.4.1`. I was able to reproduce using this version of virtualenv. I'm not really sure what's going on here but at a wild guess, maybe the older version of virtualenv is copying in an older version of the Python setuptools stuff or somehow failing to mark built-in packages such as configparser as installed.

As far as I can tell, this version of virtualenv is not required by anything in this package. Is this some quirk of how we build AMIs? If necessary, I bet we could work around this by using the old virtualenv to pip install the new virtualenv, and use that to construct the actual virtualenv that we install the package to. But maybe we can just install a modern virtualenv directly without having to jump through these hoops?

**Comment by bqbn** [comment]
@glasserc thank you for looking into this.

I later realized that the actual virtualenv version was v15.1.0.

And it also turns out virtualenv is not the problem, the real issue turns out to be that newer configparser doesn't support old setupstools any more. Since we don't pin the configparser version, it requires setuptools v30.4 and up.

https://github.com/mozilla-services/svcops/pull/1427/commits/1963ca4e005aa52fbfecc16f57b700e72dbaddbb seems to have fixed this issue for us.
